gbpservice/neutron/nsf/configurator/modules/firewall.py

```diff
@@ -0,0 +1,347 @@
+                   % (os.getpid(), ev.id, const.FIREWALL_RPC_TOPIC))
+            LOG.debug(msg)
+
+            driver = self._get_driver(ev.data)
+
+            if ev.id == 'CONFIGURE_FW':
+                driver.configure_firewall(ev)
+            elif ev.id == 'UPDATE_FW':
+                driver.update_firewall(ev)
+            elif ev.id == 'DELETE_FW':
+                driver.delete_firewall(ev)
+            else:
+                msg = ("Wrong call to configure firewall.")
+                LOG.error(msg)
+                raise Exception(msg)
+        except Exception as err:
+            LOG.error("Failed to perform the operation: %s. %s"
+                      % (ev.id, str(err).capitalize()))
+        finally:
+            self._sc.event_done(ev)
+
+
+def _create_rpc_agent(sc, topic, manager):
+    return RpcAgent(sc,
+                    host=cfg.CONF.host,
+                    topic,
+                    manager)
+
+
+def rpc_init(sc, conf):
+    fwrpcmgr = FwaasRpcReceiver(conf, sc)
+    vyosfwrpcmgr = FwGenericConfigRpcReceiver(conf, sc)
+
+    fwagent = _create_rpc_agent(sc, const.FIREWALL_RPC_TOPIC, fwrpcmgr)
+    vyosfwagent = _create_rpc_agent(sc, const.VYOS_FIREWALL_RPC_TOPIC,
+                                    vyosfwrpcmgr)
+
+    sc.register_rpc_agents([fwagent, vyosfwagent])
+
+
+def events_init(sc, drivers):
+    evs = [
+        Event(id='CONFIGURE_FW', handler=FwaasHandler(sc, drivers)),
+        Event(id='UPDATE_FW', handler=FwaasHandler(sc, drivers)),
+        Event(id='DELETE_FW', handler=FwaasHandler(sc, drivers)),
+
+        Event(id='CONFIGURE_INTERFACES', handler=FwGenericConfigHandler(
+                                                                sc, drivers)),
+        Event(id='CLEAR_INTERFACES', handler=FwGenericConfigHandler(
+                                                                sc, drivers)),
+        Event(id='CONFIGURE_LICENSE', handler=FwGenericConfigHandler(
+                                                                sc, drivers)),
+        Event(id='RELEASE_LICENSE', handler=FwGenericConfigHandler(
+                                                                sc, drivers)),
+        Event(id='CONFIGURE_SOURCE_ROUTES', handler=FwGenericConfigHandler(
+                                                                sc, drivers)),
+        Event(id='DELETE_SOURCE_ROUTES', handler=FwGenericConfigHandler(
+                                                                sc, drivers)),
+        Event(id='ADD_PERSISTENT_RULE', handler=FwGenericConfigHandler(
+                                                                sc, drivers)),
+        Event(id='DEL_PERSISTENT_RULE', handler=FwGenericConfigHandler(
+                                                                sc, drivers))]
+    sc.register_events(evs)
+
+
+def load_drivers():
+    ''' Create object of firewall drivers
+    '''
+    drivers = {"vyos_fwaas": 'FwaasDriver',
```

Comment:
TODO: We need to make this work by dynamic class detection from the driver directory and instantiate objects out of it.